File: clients/python2_client/client/core/strategy.py
from core.base_strategy import BaseStrategy


class Strategy(BaseStrategy):
    def on_tick(self, my_elevators, my_passengers, enemy_elevators, enemy_passengers):
        for elevator in my_elevators:
            passengers = [p for p in enemy_passengers if p.state < 5]
            for p in passengers:
                if elevator.state != 1:
                    elevator.go_to_floor(p.from_floor)
                if elevator.floor == p.from_floor:
                    p.set_elevator(elevator)
            if len(elevator.passengers) > 0 and elevator.state != 1:
                elevator.go_to_floor(elevator.passengers[0].dest_floor)

File: baseline/php7_client/strategy.php
<?php

require('base_strategy.php');

class Strategy extends BaseStrategy
{

    /**
     * Process strategy turn
     * @param Passenger[] $myPassengers
     * @param Elevator[] $myElevators
     * @param Passenger[] $enemyPassengers
     * @param Elevator[] $enemyElevators
     */
    public function onTick(&$myPassengers, &$myElevators, &$enemyPassengers, &$enemyElevators)
    {
        foreach ($myElevators as $elevator) {
            foreach ($myPassengers as $passenger) {
                if ($passenger->state < 5) {
                    if ($elevator->state != 1) {
                        $elevator->goToFloor($passenger->fromFloor);
                    }
                    if ($elevator->floor == $passenger->fromFloor) {
                        $passenger->setElevator($elevator);
                    }
                }
                if (count($elevator->passengers) > 0 && $elevator->state != 1) {
                    $elevator->goToFloor($elevator->passengers[0]->destFloor);
                }
            }
        }
    }

}
